### PR TITLE
add rgb25 stl + move/add common types

### PR DESCRIPTION
**Pull request by zoedberg**
As agreed on last call here the code for RGB25 interface and related updates (see https://github.com/LNP-BP/LNPBPs/issues/161).

Related PRs:
- https://github.com/RGB-WG/rgb-schemata/pull/10
- https://github.com/LNP-BP/LNPBPs/pull/163

### CODE REVIEW

File: std/src/bin/rgb-stl.rs [hunk 1]
```diff
@@ -67,6 +67,21 @@ fn main() {
...
+
+    rgb25_stl()
+        .serialize(
+            format,
+            dir.as_ref(),
+            "0.1.0",
+            Some(
+                "
+  Description: Types for RGB25 interface
+  Author: Dr Maxim Orlovsky <orlovsky@lnp-bp.org>
```
Comment (dr-orlovsky) [inline std/src/bin/rgb-stl.rs:83]:
```suggestion
  Author: Zoe Faltiba
```

File: std/src/interface/rgb25.rs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+pub fn rgb25_stl() -> TypeLib { _rgb25_stl().expect("invalid strict type RGB25 library") }
+
+pub fn rgb25() -> Iface {
+    let types = StandardTypes::with(rgb25_stl());
+
+    Iface {
+        version: VerNo::V1,
+        name: tn!("RGB25"),
+        global_state: tiny_bmap! {
+            fname!("name") => GlobalIface::required(types.get("RGBContract.Name")),
```
Comment (dr-orlovsky) [inline std/src/interface/rgb25.rs:78]: Maybe we also need an optional ticker?

File: std/src/interface/rgb25.rs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            fname!("details") => GlobalIface::optional(types.get("RGBContract.Details")),
+            fname!("precision") => GlobalIface::required(types.get("RGBContract.Precision")),
+            fname!("data") => GlobalIface::required(types.get("RGBContract.ContractData")),
+            fname!("created") => GlobalIface::required(types.get("RGBContract.Timestamp")),
+            fname!("issuedSupply") => GlobalIface::required(types.get("RGBContract.Amount")),
+        },
+        assignments: tiny_bmap! {
+            fname!("assetOwner") => AssignIface::private(OwnedIface::Amount, Req::NoneOrMore),
```
Comment (dr-orlovsky) [inline std/src/interface/rgb25.rs:86]: See my comment in LNPBP-25 standard
```suggestion
            fname!("assetOwner") => AssignIface::private(OwnedIface::Amount, Req::OneOrMore),
```

File: std/src/interface/rgb25.rs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            global: tiny_bmap! {
+                fname!("name") => ArgSpec::required(),
+                fname!("details") => ArgSpec::optional(),
+                fname!("precision") => ArgSpec::required(),
+                fname!("data") => ArgSpec::required(),
+                fname!("created") => ArgSpec::required(),
+                fname!("issuedSupply") => ArgSpec::required(),
+            },
+            assignments: tiny_bmap! {
+                fname!("assetOwner") => ArgSpec::many(),
```
Comment (dr-orlovsky) [inline std/src/interface/rgb25.rs:100]:
```suggestion
                fname!("assetOwner") => ArgSpec::non_empty(),
```

**Review by dr-orlovsky — COMMENTED** [review]
Can you pls rebase on top of master - I merged PR with a lot of fixes and now this conflicts with them

File: std/src/interface/rgb25.rs (hunk 2 above)
Comment (zoedberg) [reply std/src/interface/rgb25.rs:78]: By adding a ticker the interface would become very similar to RGB20, we prefer to keep this interface without ticker, indicating that RGB25 assets are not meant to represent "stocks", they are just collectible digital assets

File: std/src/bin/rgb-stl.rs [hunk 5]
```diff
@@ -80,7 +95,7 @@ fn main() {
             Some(
                 "
   Description: RGB standard library
-  Author: Dr Maxim Orlovsky <orlovsky@lnp-bp.org>
+  Author: Zoe Faltibà
```
Comment (dr-orlovsky) [inline std/src/bin/rgb-stl.rs:98]: You missed the line, this should go above to RGB-25 :D

Comment (dr-orlovsky) [reply std/src/bin/rgb-stl.rs:98]:
```suggestion
  Author: Dr Maxim Orlovsky <orlovsky@lnp-bp.org>
```

**Review by dr-orlovsky — APPROVED** [review]
LGTM. Do you plan to add burn procedure - or let's leave it to another PR?

File: std/src/bin/rgb-stl.rs (hunk 5 above)
Comment (zoedberg) [reply std/src/bin/rgb-stl.rs]: oops sorry :) going to fix it

**Comment by zoedberg** [comment]
@dr-orlovsky PR should be ready to be merged now

**Review by dr-orlovsky — APPROVED** [review]
ACK d9081e4c79fa6b586ff19b30fa4002fd45dcdca7

**Pull request merged**